src/promptflow/promptflow/_core/openai_injector.py

```diff
@@ -29,28 +31,40 @@ def wrapper(f):
 
         @functools.wraps(f)
         def wrapped_method(*args, **kwargs):
-            if not Tracer.active():
-                return f(*args, **kwargs)
-
-            all_kwargs = {**{k: v for k, v in zip(sig.keys(), args)}, **kwargs}
-            for key in args_to_ignore:
-                all_kwargs.pop(key, None)
             name = f.__qualname__ if not f.__module__ else f.__module__ + "." + f.__qualname__
-            trace = Trace(
-                name=name,
-                type=trace_type,
-                inputs=all_kwargs,
-                start_time=datetime.utcnow().timestamp(),
-            )
-            Tracer.push(trace)
-            try:
-                result = f(*args, **kwargs)
-            except Exception as ex:
-                Tracer.pop(error=ex)
-                raise
-            else:
-                result = Tracer.pop(result)
-            return result
+
+            with OpenTelemetryTracer.start_as_current_span(name):
```

Comment:
We'd better avoid adding OpenTelemetryTracer here, as we already have logic for the existing `Trace` class.
We can add OpenTelemetryTracer calls inside `Trace` and then tool/llm/function spans can be supported automatically.

Reply:
actually the opentelemetry calls can not be fitted in the current `Tracer`. If the span is explicitly started/ended in different functions, the auto-managed context from open telemetry will be lost.  To get the parent-child relations handled correctly, the span should be logged in an uninterrupted context, by using either `with` statement or a function decorator.